web/src/test/java/com/epam/rft/atsy/web/controllers/rest/CandidateApplicationControllerTest.java

```diff
@@ -0,0 +1,180 @@
+    given(messageSource.getMessage(Matchers.anyString(), Matchers.any(Object[].class),
+        Matchers.any(Locale.class))).willReturn(LOCALIZED_STATE_TYPE);
+  }
+
+  @Test
+  public void loadApplicationsShouldRespondWithEmptyCollectionWhenThereAreNoApplications()
+      throws Exception {
+    given(statesHistoryService.getCandidateApplicationsByCandidateId(CANDIDATE_ID)).willReturn(
+        Collections.emptyList());
+
+    mockMvc.perform(
+        get(REQUEST_URL + CANDIDATE_ID.toString()).accept(MediaTypes.APPLICATION_JSON_UTF8))
+        .andExpect(status().isOk())
+        .andExpect(content().contentType(MediaTypes.APPLICATION_JSON_UTF8))
+        .andExpect(jsonPath("$").isArray())
+        .andExpect(jsonPath("$").isEmpty());
+
+    then(statesHistoryService).should().getCandidateApplicationsByCandidateId(CANDIDATE_ID);
+
+    verifyZeroInteractions(messageSource);
+  }
+
+  @Test
+  public void loadApplicationsShouldRespondWithSingleElementArrayWhenThereIsOnlyOneApplication()
+      throws Exception {
+    given(statesHistoryService.getCandidateApplicationsByCandidateId(CANDIDATE_ID))
+        .willReturn(Collections.singletonList(developerApplicationDto));
+
+    mockMvc.perform(
+        get(REQUEST_URL + CANDIDATE_ID.toString()).accept(MediaTypes.APPLICATION_JSON_UTF8))
+        .andExpect(status().isOk())
+        .andExpect(content().contentType(MediaTypes.APPLICATION_JSON_UTF8))
+        .andExpect(jsonPath("$").isArray())
+        .andExpect(jsonPath("$[0]").exists())
+        .andExpect(jsonPath("$[0].lastStateId", equalTo(LAST_STATE_ID.intValue())))
+        .andExpect(jsonPath("$[0].applicationId", equalTo(APPLICATION_ID.intValue())))
+        .andExpect(jsonPath("$[0].positionName", equalTo(DEVELOPER_POSITION)))
+        .andExpect(jsonPath("$[0].creationDate", equalTo(CREATION_DATE)))
+        .andExpect(jsonPath("$[0].modificationDate", equalTo(MODIFICATION_DATE)))
+        .andExpect(jsonPath("$[0].stateType", equalTo(LOCALIZED_STATE_TYPE)))
+        .andExpect(jsonPath("$[1]").doesNotExist());
+
+    then(statesHistoryService).should().getCandidateApplicationsByCandidateId(CANDIDATE_ID);
+
+    then(messageSource).should()
+        .getMessage(eq(APPLICATION_STATE + RAW_STATE_TYPE), eq(new Object[]{RAW_STATE_TYPE}),
+            any(Locale.class));
+  }
+
+  @Test
+  public void loadApplicationsShouldRespondWithTwoElementArrayWhenThereAreTwoApplications()
+      throws Exception {
+    given(statesHistoryService.getCandidateApplicationsByCandidateId(CANDIDATE_ID))
+        .willReturn(candidateApplicationCollection);
+
+    mockMvc.perform(
+        get(REQUEST_URL + CANDIDATE_ID.toString()).accept(MediaTypes.APPLICATION_JSON_UTF8))
+        .andExpect(status().isOk())
+        .andExpect(content().contentType(MediaTypes.APPLICATION_JSON_UTF8))
+        .andExpect(jsonPath("$").isArray())
+        .andExpect(jsonPath("$[0]").exists())
+        .andExpect(jsonPath("$[0].lastStateId", equalTo(LAST_STATE_ID.intValue())))
```

Comment:
Please use e.g. developerApplicationDto.getLastStateId instead of the constants here, as we want to check if the JSON contains values from that are in the objects.  

The result is the same, but it seem more semantically correct.